H2MLauncher.UI/ViewModels/ServerTabViewModel.cs

```diff
@@ -28,23 +72,62 @@ public partial class ServerTabViewModel<T> : ObservableObject where T : IServerV
 
     [ObservableProperty]
     [NotifyCanExecuteChangedFor(nameof(JoinServerCommand))]
-    private T? _selectedServer;
+    private TServerViewModel? _selectedServer;
+
+    private readonly CollectionViewSource _collectionViewSource;
+
+    public ObservableCollection<TServerViewModel> Servers { get; } = [];
+    ICollection<TServerViewModel> IServerTabViewModel<TServerViewModel>.Servers => Servers;
```

Comment:
The other two are necessary explicit implementations so there more generic Interface can Access the list. (As i am looking at this maybe the second one can be implicitly implemented by the ObservableCollection). The IServerTabViewModel.Servers needs to be an IEnumerable because only it is covariant (implicitly castable to from ICollection)

Reply:
Unfortunately does not work, needs to be the same return type for interfaces